src/flamegraph.cpp

```diff
@@ -546,14 +546,13 @@ struct SearchResults
     qint64 directCost = 0;
 };
 
-SearchResults applySearch(FrameGraphicsItem* item, const QString& searchValue)
+SearchResults applySearch(FrameGraphicsItem* item, const QRegularExpression& expression)
 {
     SearchResults result;
-    if (searchValue.isEmpty()) {
+    if (expression.pattern().isEmpty()) {
```

Comment:
I'm not sure if this should be `!expression.isValid() || expression.pattern().isEmpty()`

Reply:
if it isn't valid we should not get here, at least not when executed from the UI (the function could possibly `assert` that or do an early-exit for the invalid expression)